src/useChat.ts

```diff
@@ -1,120 +1,162 @@
+ * A hook that exposes:
+ *  - sendMessage: a request utility for sending a text string to the query API
+ *  - activeMessage: an object representing a message that the platform has not completed responding to yet
+ *  - messageHistory: an array of previous messages
+ *  - isLoading: a boolean indicating a pending response from the platform
+ *  - isStreamingResponse: a boolean indicating that the platform is currently streaming data to the client
+ *  - startNewConversation: a utility method for clearing the chat context
+ *  - hasError: a boolean indicating an error was encountered while sending a message to the platform
  */
-export const useChat = (customerId: string, corpusIds: string[], apiKey: string) => {
+export const useChat = (customerId: string, corpusIds: string[], apiKey: string, useStreaming: boolean = true) => {
   const [messageHistory, setMessageHistory] = useState<ChatTurn[]>([]);
   const recentQuestion = useRef<string>("");
-  const [recentAnswer, setRecentAnswer] = useState<ChatTurn | null>();
+  const [activeMessage, setActiveMessage] = useState<ChatTurn | null>(null);
   const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [conversationId, setConversationId] = useState<string | undefined>();
+  const [isStreamingResponse, setIsStreamingResponse] = useState<boolean>(false);
+  const [conversationId, setConversationId] = useState<string | null>(null);
   const [hasError, setHasError] = useState<boolean>(false);
+
   const getLanguage = (languageValue?: string): SummaryLanguage => (languageValue ?? "eng") as SummaryLanguage;
 
   const sendMessage = async ({ query, isRetry = false }: { query: string; isRetry?: boolean }) => {
     if (isLoading) return;
-    recentQuestion.current = query;
-
-    // If this isn't a retry, optimistically add a placeholder entry to the chat history.
-    // We'll replace this later on with a copy from the server.
-    if (!isRetry) {
-      setMessageHistory((messageHistory) => {
-        return [
-          ...messageHistory,
-          {
-            id: "placeholder-message-id",
-            question: query,
-            answer: "",
-            results: []
-          }
-        ];
-      });
-    } else {
+    if (isRetry) {
       setHasError(false);
     }
 
+    setActiveMessage(null);
+    recentQuestion.current = query;
+
+    // Optimistically add a placeholder entry as the active message.
+    // We'll replace this later on with data from the server.
+    setActiveMessage({
+      id: "placeholder-message-id",
+      question: query,
+      answer: "",
+      results: []
+    });
+
     const baseSearchRequestParams = {
       filter: "",
       queryValue: query,
       rerank: true,
-      rerankNumResults: 50,
+      rerankNumResults: 10,
       rerankerId: 272725718,
       rerankDiversityBias: 0.3,
-      hybridNumWords: 2,
-      hybridLambdaLong: 0.0,
-      hybridLambdaShort: 0.1,
```

Comment:
After a convo with Ofer, I updated stream-query-client to just take an optional lambda value which defaults to 0.025. He mentioned the notion of long and short values + number of words was something he came up specifically for vectara-answer.